ObjectiveGit/GTTreeEntry.m

```diff
@@ -39,6 +39,7 @@
 
 @interface GTTreeEntry ()
 @property (nonatomic, assign, readonly) const git_tree_entry *git_tree_entry;
+@property (nonatomic, assign, readonly) git_tree_entry* entry_to_free;
```

Comment:
I thougt about that, but that would cause interface changes that might break other projects.
Furthermore, the existing modifier is currently needed since methods like "git_tree_entry_byname" return "const git_tree_entry *". One solution might be to copy those tree entries before adding them to GTTreeEntry.